frontend/src/layouts/cameras/index.tsx

```diff
@@ -25,55 +28,155 @@ interface DeviceRemovedInfo {
-        setDevices((prevDevices: Device[]) => {
-            return [...prevDevices, device];
-        });
+        if (devices.find((dev) => dev.bus_info === device.bus_info)) {
+            updateDevice(device);
+        } else {
+            setDevices((prevDevices: Device[]) => {
+                return [...prevDevices, device];
+            });
+        }
     };
 
     const removeDevice = (bus_info: string): void => {
         setDevices((prevDevices: Device[]) => {
+            for (let device of prevDevices) {
+                if (device.bus_info == bus_info && device.follower) {
+                    console.log(device.follower);
+                    removeLeaderUpdate(device.follower, prevDevices);
+                }
+            }
             return prevDevices.filter((device) => {
                 return device.bus_info != bus_info;
             });
         });
     };
 
+    const updateDevice = (device: Device) => {
+        setDevices((prevDevices) => {
+            let newDevices = [...prevDevices];
+            newDevices = newDevices.filter(
+                (dev) => dev.bus_info != device.bus_info
+            );
+            newDevices.push(device);
+            return newDevices;
+        });
+    };
+
     useEffect(() => {
         // Code to run once when the component is defined
         getDevices().then((devices) => {
             console.log("Devices: ", devices);
             addDevices(devices);
         });
 
-        websocket.addEventListener("message", (e) => {
+        const socketCallback = (e) => {
             let message = deserializeMessage(e.data);
-            console.log(message.data);
             switch (message.event_name) {
-                case "device_added":
+                case "device_added": {
+                    let dev = message.data as Device;
+                    enqueueSnackbar(
+                        `Device added: ${dev.bus_info} - ${dev.nickname || dev.device_type}`,
+                        {
+                            variant: "info",
+                        }
+                    );
                     addDevice(message.data as Device);
                     break;
-                case "device_removed":
+                }
+                case "device_removed": {
+                    let dev = message.data as DeviceRemovedInfo;
+                    enqueueSnackbar(`Device removed: ${dev.bus_info}`, {
+                        variant: "info",
+                    });
                     removeDevice((message.data as DeviceRemovedInfo).bus_info);
                     break;
+                }
+                case "gst_error":
+                    let gstErrorMessage = message.data as GstErrorMessage;
+                    console.log(gstErrorMessage);
+                    enqueueSnackbar(
+                        `GStreamer Error Occurred: ${gstErrorMessage.bus_info} - This is likely a known issue with the kernel, please read our docs site for more details`,
```

Comment:
do we want to include a link?

Reply:
We do not yet have a link to the kernel issue documentation. We need to make that